### PR TITLE
add py-gemini 

### PR DESCRIPTION
**Pull request by manifestoso**
new bio tools py-gemini

### CODE REVIEW

File: var/spack/repos/builtin/packages/bedtools2/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/bedtools2/package.py]: Can you contact the developers and see why the checksum of old software was changed?

File: var/spack/repos/builtin/packages/py-pybedtools/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pybedtools/package.py]: It looks like pandas and numpy are now optional. Also, for setuptools we need `@0.6c5:`

File: var/spack/repos/builtin/packages/sra-tools/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sra-tools/package.py]:
```suggestion
        install_tree('example', prefix.example)
```

File: var/spack/repos/builtin/packages/sra-tools/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sra-tools/package.py]:
```suggestion
        install_tree('schema', prefix.schema)
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by adamjstewart** [comment]
I don't actually see `py-gemini` in this PR...

File: var/spack/repos/builtin/packages/bedtools2/package.py
Comment (manifestoso) [reply var/spack/repos/builtin/packages/bedtools2/package.py]: @adamjstewart Thx for point this out. The `URL` to pull the tarball is __https://github.com/arq5x/bedtools2/archive/v2.26.0.tar.gz__
The `spack checksum bedtools2` pull checksum from __" releases/download/v2.26.0/bedtools-2.29.1.tar.gz"__ handle. 

Do we have a batch method to checksum with `spack`?

File: var/spack/repos/builtin/packages/py-pybedtools/package.py
Comment (manifestoso) [reply var/spack/repos/builtin/packages/py-pybedtools/package.py]: Thx @adamjstewart, `setuptools` version updated. 

It seems both `pandas` and `numpy` still listed under `requirement.txt` see 
[here](https://github.com/daler/pybedtools/blob/master/requirements.txthttps://github.com/daler/pybedtools/blob/master/requirements.txt).

File: var/spack/repos/builtin/packages/sra-tools/package.py
Comment (manifestoso) [reply var/spack/repos/builtin/packages/sra-tools/package.py]: My ignorance, fixed in #16947. Thx @adamjstewart

**Review by manifestoso — COMMENTED** [review]
Git history clean up, this stack is now independent from #16947. Thx @adamjstewart 

LGTM

**Comment by manifestoso** [comment]
> 
> 
> I don't actually see `py-gemini` in this PR...

My bad, now included. Thx @adamjstewart

File: var/spack/repos/builtin/packages/bedtools2/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/bedtools2/package.py]: > Do we have a batch method to checksum with spack?

The batch method _is_ `spack checksum`, but it isn't perfect. If multiple URLs exist at the same location for the same version, it can do the wrong thing.

File: var/spack/repos/builtin/packages/py-pybedtools/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pybedtools/package.py]: Ignore `requirements.txt`, see https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#python-dependencies

Comment (scheibelp) [reply var/spack/repos/builtin/packages/py-pybedtools/package.py]: > Also, for setuptools we need `@0.6c5:`

That doesn't seem to correspond with any of our `py-setuptools` versions, where did that requirement come from?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pybedtools/package.py]: From the `setup.py`

File: var/spack/repos/builtin/packages/py-pybedtools/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pybedtools/package.py]:
```suggestion
    depends_on('py-setuptools@0.6c5:', type='build')
```

File: var/spack/repos/builtin/packages/py-pybedtools/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pybedtools/package.py]: I think `py-cython` is only needed when you build from source, the release tarball shouldn't need this dependency

File: var/spack/repos/builtin/packages/py-gemini/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gemini/package.py]:
```suggestion
    depends_on('python@2.5:2.8', type=('build', 'run'))
```
It looks like it supports Python 2.5 and 2.6 too

File: var/spack/repos/builtin/packages/py-gemini/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gemini/package.py]: These four dependencies don't look like they are needed. Compilers aren't dependencies at the moment, so if this requires an older version of GCC (or another compiler to build) you can add:
```python
conflicts('%gcc@7:', msg='py-gemini doesn't work with GCC 7+')
```
I assume git and wget are only needed to download the source code, not needed to build the package.

File: var/spack/repos/builtin/packages/py-gemini/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gemini/package.py]: Newer versions of setuptools don't work?

File: var/spack/repos/builtin/packages/py-gemini/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gemini/package.py:34]: I'm seeing a lot of missing dependencies and incorrect dependency versions:
```
numpy>=1.7.1
inheritance>=0.1.3
geneimpacts>=0.1.3
cython>=0.22.1
sqlalchemy>=1
pysam>=0.6
cyvcf2>=0.7.2
PyYAML>=3.10
pybedtools>=0.6.2
jinja2>=2.7.1
networkx>=1.10
bottle>=0.11.6
ipyparallel>=4.0
ipython-cluster-helper>=0.5.1
bx-python>=0.7.1
pandas>=0.11.0
openpyxl>=1.6.1,<2.0.0
scipy>=0.12.0
Unidecode>=0.04.14
cyordereddict==0.2.2
bcolz>=0.11.3
numexpr>=2.4.3
gemini==0.19.1
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by adamjstewart** [comment]
I know this is annoying, but it would be much easier for me to review this PR if you create separate PRs for each package. Also see https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#url

**Comment by manifestoso** [comment]
> 
> 
> I know this is annoying, but it would be much easier for me to review this PR if you create separate PRs for each package. Also see https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#url

@adamjstewart Surely can. However, given that it's all dependencies, it may be hard to keep track. What I can do though is that split it up into 2 packages and we get it done once at a time. See what's your thought.

**Comment by adamjstewart** [comment]
I think it's fine to split them all into separate PRs. Spack has dependency checks now to prevent a package from being merged until its dependencies are merged. You'll just need to rebase/merge once the dependencies get merged.

**Comment by manifestoso** [comment]
> 
> 
> I think it's fine to split them all into separate PRs. Spack has dependency checks now to prevent a package from being merged until its dependencies are merged. You'll just need to rebase/merge once the dependencies get merged.

@adamjstewart SGTM. That's the plan.

File: var/spack/repos/builtin/packages/py-pybedtools/package.py
Comment (manifestoso) [reply var/spack/repos/builtin/packages/py-pybedtools/package.py]: Fixed here.

**Review by manifestoso — COMMENTED** [review]
checksum converges to `source`

File: var/spack/repos/builtin/packages/grabix/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/grabix/package.py]: Compilers are not currently modeled as dependencies. I think what you want is:
```python
conflicts('%gcc@7:', msg='grabix cannot be compiled with newer versions of GCC')
```

File: var/spack/repos/builtin/packages/grabix/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/grabix/package.py]: I'm not sure if this line is worth adding since you are listed under `maintainers`

File: var/spack/repos/builtin/packages/py-gemini/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gemini/package.py]: Same here

File: var/spack/repos/builtin/packages/py-gemini/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gemini/package.py]: Build dependencies add `prefix.bin` to the `PATH`. `zlib` doesn't install anything to `prefix.lib`, so this dependency isn't being used right now. I would either remove it or make it a link dep.

File: var/spack/repos/builtin/packages/py-gemini/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gemini/package.py]: Same comment about compilers as dependencies

File: var/spack/repos/builtin/packages/py-gemini/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gemini/package.py]: Where did you find these two dependencies?

File: var/spack/repos/builtin/packages/py-gemini/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gemini/package.py]:
```suggestion
    depends_on('py-bottle@0.11.6:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-gemini/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gemini/package.py]:
```suggestion
    depends_on('py-openpyxl@1.6.1:1.999', type=('build', 'run'))
```
Version ranges are inclusive

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/grabix/package.py
Comment (manifestoso) [reply var/spack/repos/builtin/packages/grabix/package.py]: Fixed.

File: var/spack/repos/builtin/packages/grabix/package.py
Comment (manifestoso) [reply var/spack/repos/builtin/packages/grabix/package.py]: Legacy header fixed.

File: var/spack/repos/builtin/packages/py-gemini/package.py
Comment (manifestoso) [reply var/spack/repos/builtin/packages/py-gemini/package.py]: Legacy deps removed

File: var/spack/repos/builtin/packages/py-gemini/package.py
Comment (manifestoso) [reply var/spack/repos/builtin/packages/py-gemini/package.py]: Fixed. Thx

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
